app/models/papi_client.rb

```diff
@@ -0,0 +1,307 @@
+        resources: resources,
+        timeout: timeout
+    )
+  end
+
+  # Instantiate actions for pipeline, which holds command line actions, docker information,
+  # and information that is passed to run_pipeline.  The Docker image that is pulled for this
+  # is hard-coded to PapiClient::INGEST_DOCKER_IMAGE
+  #
+  # * *params*
+  #   - +commands+: (Array<String>) => An array of commands to run inside the container
+  #   - +environment+: (Hash) => Hash of key/value pairs to set as the container env
+  #   - +flags+: (Array<String>) => An array of flags to apply to the action
+  #   - +image_uri+: (String) => GCR Docker image to pull, defaults to PapiClient::INGEST_DOCKER_IMAGE
+  #   - +labels+: (Hash) => Hash of labels to associate with the action
+  #   - +timeout+: (String) => Maximum runtime of action
+  #
+  #  * *return*
+  #   - (Google::Apis::GenomicsV2alpha1::Action)
+  def create_actions_object(commands: [], environment: {}, flags: [], labels: {}, timeout: nil)
+    Google::Apis::GenomicsV2alpha1::Action.new(
+        commands: commands,
+        environment: environment,
+        flags: flags,
+        image_uri: INGEST_DOCKER_IMAGE,
+        labels: labels,
+        timeout: timeout
+    )
+  end
+
+  # Instatiate a resources object to tell where to run a pipeline
+  #
+  # * *params*
+  #   - regions: (Array<String>) => An array of GCP regions allowed for VM allocation
+  #
+  # * *return*
+  #   - (Google::Apis::GenomicsV2alpha1::Resources)
+  def create_resources_object(regions:)
+    Google::Apis::GenomicsV2alpha1::Resources.new(
+         project_id: COMPUTE_PROJECT,
+         regions: regions,
+         virtual_machine: self.create_virtual_machine_object
+    )
+  end
+
+  # Instantiate a VM object to specify in resources.  Assigns the portal service account to the VM
+  # to manage permissions
+  #
+  # * *params*
+  #   - +machine_type+ (String) => GCP VM machine type (defaults to 'n1-standard-1')
+  #   - +preemptible+ (Boolean) => Indication of whether VM can be preempted (defaults to false)
+  # * *return*
+  #   - (Google::Apis::GenomicsV2alpha1::VirtualMachine)
+  def create_virtual_machine_object(machine_type: 'n1-standard-1', preemptible: false)
+    Google::Apis::GenomicsV2alpha1::VirtualMachine.new(
+        machine_type: machine_type,
+        preemptible: preemptible,
+        service_account: Google::Apis::GenomicsV2alpha1::ServiceAccount.new(email: self.issuer, scopes: GOOGLE_SCOPES)
+    )
+  end
+
+  # Determine command line to pass to ingest based off of file & action requested
+  #
+  # * *params*
+  #   - +study_file+ (StudyFile) => StudyFile to be ingested
+  #   - +action+ (String/Symbol) => Action to perform on ingest
+  #
+  # * *return*
+  #   - (Array) Command Line, in Docker "exec" format
+  #
+  # * *raises*
+  #   - (ArgumentError) => The requested StudyFile and action do not correspond with each other, or cannot be run yet
+  def get_command_line(study_file:, action:)
```

Comment:
Succinct!

I'll keep an eye on this -- I'm interested to see how our forthcoming integration validation commands will relate to this method.